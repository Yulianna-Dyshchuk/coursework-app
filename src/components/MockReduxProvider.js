import React from "react";
import { combineReducers, createStore } from "redux";
import { Provider } from "react-redux";

const store = createStore(combineReducers({ categories: () => ([]) }));

export const MockReduxProvider = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
