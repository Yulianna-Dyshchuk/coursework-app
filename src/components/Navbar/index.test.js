import React from "react";
import { shallow } from "enzyme";
import { Navbar } from "./index";
import { MockReduxProvider } from "../MockReduxProvider";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({ pathname: "localhost:3000/" })
}));

describe("Navbar component", () => {
  it("should render correctly without props", () => {
    const component = shallow(
      <MockReduxProvider>
        <Navbar />
      </MockReduxProvider>
    );

    expect(component).toMatchSnapshot();
  });
});
