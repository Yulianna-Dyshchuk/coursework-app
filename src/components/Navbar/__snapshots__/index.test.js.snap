// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correctly without props 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Navbar />
</Provider>
`;
